### DISCUSSION TITLE
Curry/Array/Method

### DISCUSSION THREAD

**Discussion by AlexShopiak** [HowProgrammingWorks/LiveQA#12] [Q&A]
1. Если мы каррируем функцию, частично применяем ее без последнего аргумента, и потом, на основе частично примененной, создаем инстансы для разных последних аргументов. Это оптимизирует код или только улучшает его читабельность? 

2. В лекции про массивы, есть метод `.every()`. В примере для него был массив  `['a', 'b', 'c']`  и 
проверка `array.every((c) => c > 'a')`. Так вот нету **точной** информации, каким образом в JS сравниваются буквы. Предпологаю, что просто сравниваются индексы букв английского алфавита.....?Либо там что-то интересеней... ?   

3. Если мы используем метод в коде, но "юзаем" лишь малую часть его возможностей. Практично ли будет  реализовывать нужную нам часть его функционала самому, с помощью циклов и разветвлений? Или это значительно ни на что не повлияет?

**Comment by tshemsedinov** [comment]
1. Ни каких оптимизаций тут нет, но может улучшать читаемость, а может и этого не делать
2. В js нет типа `char`, а у нас тут сравниваются строки, но они сравниваются по буквам и конечно по кодам в ASCII таблице
3. Вопрос не ясен, нельзя использовать только часть возможностей метода
